### PR TITLE
initiate a model for on demand running validation

### PR DESCRIPTION
**Pull request by mike-stern**

### CODE REVIEW

File: .github/workflows/runSchematronValidation.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      - name: Checkout submodules
+        uses: textbook/git-checkout-submodule-action@2.1.1
+        with:
+          remote: true
+
+      # Install Saxon HE to /tmp
+      - name: Install Saxon HE
+        run: |
+          echo "Installing Saxon"
+          mkdir -p /tmp/saxon
```
Comment (ohsh6o) [inline .github/workflows/runSchematronValidation.yml:31]: So we are moving away from the `mvn` approach and caching things in `~/.m2` within the context of the container user?

[For reference](https://github.com/ohsh6o/skyrim-schematron-tutorial/blob/main/.github/workflows/report.yml#L37-L41)

File: .github/workflows/runSchematronValidation.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+          echo "Dowloading Saxon"
+          export SAXON_CP=/tmp/saxon/Saxon-HE-10.2.jar
+          wget -O "${SAXON_CP}" https://repo1.maven.org/maven2/net/sf/saxon/Saxon-HE/10.2/Saxon-HE-10.2.jar
```
Comment (ohsh6o) [inline .github/workflows/runSchematronValidation.yml:34]: Do we want to consider caching the file and keying it in the cache? This way, if we will run many times, we will significantly cut down on DL time, `mvn` or `wget` regardless.

[For reference](https://github.com/ohsh6o/skyrim-schematron-tutorial/blob/main/.github/workflows/report.yml#L29-L35)

File: .github/workflows/runSchematronValidation.yml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+          # compute name without .sch
+          qualifiedSchematronName=${{ schematronFileName }}
+          schematronName=${qualifiedSchematronName##*/}
+          schematronRoot=${schematronName%.*}
+          rm -rf target/*.xsl;
+          java -cp "${SAXON_CP}" net.sf.saxon.Transform \
+          -o:target/"${schematronRoot}".xsl \
+          -s:"${qualifiedSchematronName}}" \
+          lib/schematron/trunk/schematron/code/iso_svrl_for_xslt2.xsl \
+          allow-foreign=true
```
Comment (ohsh6o) [inline .github/workflows/runSchematronValidation.yml:50]: Should we consider setting up a `$SAXON_OPTS` environment variable so we can change it for all commands like we had before?

For reference:
- [Variable declaration](https://github.com/18F/fedramp-automation/blob/4bc1ec9e0c018b536165175b07a6fd8d507b0c7c/resources/validations/bin/validate_with_schematron.sh#L17)
- [Variable use](https://github.com/18F/fedramp-automation/blob/4bc1ec9e0c018b536165175b07a6fd8d507b0c7c/resources/validations/bin/validate_with_schematron.sh#L44)

**Review by ohsh6o — CHANGES_REQUESTED** [review]
I like what I see, but can we also save the reports? My only general question: can we [store the resulting output](https://github.com/ohsh6o/skyrim-schematron-tutorial/blob/main/.github/workflows/report.yml#L65-L74) for demos and stuff?

**Comment by mike-stern** [comment]
This is the simplest version of the action. I agree with all the input, i am not actually ready for a review, but i need to merge to master in order to round it out.
(unless there is a way to see a action that exists only on a branch)

**Comment by ohsh6o** [comment]
> (unless there is a way to see a action that exists only on a branch)

This "feature" has also been driving me nuts, so now I want to examine why this is! lol

**Review by ohsh6o — APPROVED** [review]
And AJ goes whoosh and [clearly did not read what you have been throwing out to me in Slack repeatedly!](https://stackoverflow.com/questions/58933155/manual-workflow-triggers-in-github-actions/62768214#62768214) lol

**Pull request merged**